python/GafferUI/PathListingWidget.py

```diff
@@ -792,33 +862,33 @@ def drawRow( self, painter, option, index ) :
 
 		QtWidgets.QTreeView.drawRow( self, painter, option, index )
 
-	def drawBranches( self, painter, rect, index ) :
+	def __drawSelectionHighlight( self, painter, rect, index ) :
 
-		# Qt has a bug whereby it double-draws the row background in the branch
-		# area, once in `drawRow()` and then once in `drawBranches()`. This causes
-		# partially-transparent backgrounds to be drawn too heavy. Refill with the
-		# default background colour before drawing our highlight and deferring the
-		# rest to Qt.
-		painter.fillRect( rect, QtGui.QColor( *(_styleColors["backgroundRaised"]) ) )
+		header = self.header()
+		descendantMatch = False
 
-		self.__drawSelectionHighlight( painter, rect, index )
+		for i in range( 0, header.count() ) :
+			columnIndex = index.siblingAtColumn( i )
+			match = columnIndex.model().data( columnIndex, QtCore.Qt.UserRole )
 
-		QtWidgets.QTreeView.drawBranches( self, painter, rect, index )
+			if match & IECore.PathMatcher.Result.DescendantMatch :
+				descendantMatch = True
 
-	def __drawSelectionHighlight( self, painter, rect, index ) :
+			activeRect = rect
 
-		match = index.model().data( index, QtCore.Qt.UserRole )
-		if match & IECore.PathMatcher.Result.ExactMatch :
-			alpha = 200
-		elif match & IECore.PathMatcher.Result.DescendantMatch :
-			alpha = 50
-		else :
-			alpha = None
+			if match & IECore.PathMatcher.Result.ExactMatch :
+				left = header.sectionViewportPosition( i )
+				width = ( header.sectionViewportPosition( i + 1 ) - left ) if ( i < header.count() - 1 ) else ( rect.right() - left )
+				activeRect = QtCore.QRectF( left, rect.top(), width, rect.height() )
+
+				color = QtGui.QColor( *(_styleColors["brightColor"]) )
+				color.setAlpha( 200 )
+				painter.fillRect( activeRect, color )
 
-		if alpha :
+		if descendantMatch :
 			color = QtGui.QColor( *(_styleColors["brightColor"]) )
-			color.setAlpha( alpha )
-			painter.fillRect( rect, color )
+			color.setAlpha( 50 )
+			painter.fillRect( activeRect, color )
```

Comment:
We should be painting `rect` here, to paint the full row. If the last element in the row is selected, then `activeRect` is just the rect for the last cell. This causes the issue below, where we lose the descendant highlighting for the top row...

![image](https://user-images.githubusercontent.com/1133871/176470455-f7bf0e43-6251-4bd5-80e4-c0be5085b1e1.png)

Reply:
Fixed in 8800d2172415d8bdcad82244c1be89edc2341afb